File: src/Qwack.Transport/TransportObjects/Instruments/Funding/TO_FloatRateIndex.cs
using ProtoBuf;
using Qwack.Transport.BasicTypes;

namespace Qwack.Transport.TransportObjects.Instruments.Funding
{
    [ProtoContract]
    public class TO_FloatRateIndex
    {
        [ProtoMember(1)]
        public DayCountBasis DayCountBasis { get; set; }
        [ProtoMember(2)]
        public DayCountBasis DayCountBasisFixed { get; set; }
        [ProtoMember(3)]
        public string ResetTenor { get; set; }
        [ProtoMember(4)]
        public string ResetTenorFixed { get; set; }
        [ProtoMember(5)]
        public string HolidayCalendars { get; set; }
        [ProtoMember(6)]
        public RollType RollConvention { get; set; }
        [ProtoMember(7)]
        public string Currency { get; set; }
        [ProtoMember(8)]
        public string FixingOffset { get; set; }
        [ProtoMember(9)]
        public string Name { get; set; }
    }
}

File: src/Qwack.Transport/GeneratedTypeScript/TO_FloatRateIndex.ts

import { DayCountBasis } from './DayCountBasis';
import { RollType } from './RollType';

export interface TO_FloatRateIndex  {
	 dayCountBasis: DayCountBasis;
	 dayCountBasisFixed: DayCountBasis;
	 resetTenor: string;
	 resetTenorFixed: string;
	 holidayCalendars: string;
	 rollConvention: RollType;
	 currency: string;
	 fixingOffset: string;
}


